modules/auxiliary/gather/memcached_extractor.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Memcached Extractor',
+      'Description'   => %q(
+        This module extracts the slabs from a memcached instance.  It then
+        finds the keys and values stored in those slabs.
+      ),
+      'Author'        => [ 'Paul Deardorff <paul_deardorff[at]rapid7.com>' ],
+      'License'       => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(11211),
+        OptInt.new('MAXKEYS', [ true, 'Maximum number of keys to be pulled from a slab', 100] )
```

Comment:
I don't know what a slab or a key is in the context of memcached, but if 100 is a sane default then I'd probably just move this to ```register_advanced_options``` if it won't need to be changed on a regular basis.